src/configuration.rs

```diff
@@ -0,0 +1,166 @@
+
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+  pub line_width: u32,
+  pub use_tabs: bool,
+  pub indent_width: u8,
+  pub new_line_kind: NewLineKind,
+  /// Formatting program to run
+  pub executable: String,
+  pub exe_dir_path: PathBuf,
+  pub stdin: bool,
+  /// Executable arguments to add
+  pub args: Vec<String>,
+  pub timeout: u32,
+}
+
+impl Configuration {
+  /// Resolves configuration from a collection of key value strings.
+  ///
+  /// # Example
+  ///
+  /// ```
+  /// use std::collections::HashMap;
+  /// use dprint_core::configuration::{resolve_global_config};
+  /// use dprint_plugin_exec::configuration::Configuration;
+  ///
+  /// let config_map = HashMap::new(); // get a collection of key value pairs from somewhere
+  /// let global_config_result = resolve_global_config(config_map, &Default::default());
+  ///
+  /// // check global_config_result.diagnostics here...
+  ///
+  /// let exec_config_map = HashMap::new(); // get a collection of k/v pairs from somewhere
+  /// let config_result = Configuration::resolve(
+  ///     exec_config_map,
+  ///     &global_config_result.config
+  /// );
+  ///
+  /// // check config_result.diagnostics here and use config_result.config
+  /// ```
+  pub fn resolve(
+    config: ConfigKeyMap,
+    global_config: &GlobalConfiguration,
+  ) -> ResolveConfigurationResult<Configuration> {
+    let mut diagnostics = vec![];
+    let mut config = config;
+
+    let resolved_config = Configuration {
+      line_width: get_value(
+        &mut config,
+        "lineWidth",
+        global_config
+          .line_width
+          .unwrap_or(DEFAULT_GLOBAL_CONFIGURATION.line_width),
+        &mut diagnostics,
+      ),
+      use_tabs: get_value(
+        &mut config,
+        "useTabs",
+        global_config
+          .use_tabs
+          .unwrap_or(DEFAULT_GLOBAL_CONFIGURATION.use_tabs),
+        &mut diagnostics,
+      ),
+      indent_width: get_value(
+        &mut config,
+        "indentWidth",
+        global_config
+          .indent_width
+          .unwrap_or(DEFAULT_GLOBAL_CONFIGURATION.indent_width),
+        &mut diagnostics,
+      ),
+      new_line_kind: get_value(
+        &mut config,
+        "newLineKind",
+        global_config
+          .new_line_kind
+          .unwrap_or(DEFAULT_GLOBAL_CONFIGURATION.new_line_kind),
+        &mut diagnostics,
+      ),
+      executable: get_value(
+        &mut config,
+        "executable",
+        String::default(),
+        &mut diagnostics,
+      ),
+      exe_dir_path: Configuration::get_exe_dir_path(get_value(
+        &mut config,
+        "execDirPath",
+        String::default(),
+        &mut diagnostics,
+      )),
+      stdin: get_value(&mut config, "stdin", false, &mut diagnostics),
+      timeout: get_value(&mut config, "timeout", 30, &mut diagnostics),
+      args: get_value(&mut config, "args", String::default(), &mut diagnostics)
+        .split(" ")
```

Comment:
This is unfortunate... I wish `ConfigKeyMap` have a string array type, then I can avoid parsing a string here and configure the plugin like:
```json
"args": [
  "-jar",
  "formatter.jar",
  "{{file_path}}"
]
```

Reply:
Yeah, this should also probably take into account quotes. I opened https://github.com/dprint/dprint-plugin-exec/issues/2

It might be nice to make dprint support nested objects in plugin config. I'm not sure it's worth the extra complexity though as providing a string here is probably not too bad. Let's see how it goes.